File: VuongVL/app/src/main/java/com/example/vuongvl/ServiceMail.java
package com.example.vuongvl;

import android.os.CountDownTimer;
import java.util.ArrayList;

public class ServiceMail {
    CountDownTimer timer;
    ArrayList<Client> word;
    ISpam iSpam;

    public ServiceMail(ISpam iSpam) {
        this.iSpam = iSpam;
    }

    public void registerSpam(final ArrayList<Client> mClient) {
        timer = new CountDownTimer(30000, 1000) {
            @Override
            public void onTick(long millisUntilFinished) {
            }

            @Override
            public void onFinish() {
                for (Client client : mClient) {
                    if (client.typeService) {
                        iSpam.spamMail(client.getName());
                    }
                }
                timer.start();
            }
        };
        timer.start();
    }
}

File: VuongVLKotlin/app/src/main/java/com/example/vuongvlkotlin/ServiceMail.kt
package com.example.vuongvlkotlin

import android.os.CountDownTimer
import java.util.*

class ServiceMail{
    var timer: CountDownTimer? = null
    var word: ArrayList<Client>? = null
    var iSpam: ISpam? = null

    fun ServiceMail(iSpam: ISpam?) {
        this.iSpam = iSpam
    }

    fun registerSpam(mClient: ArrayList<Client>) {
        timer = object: CountDownTimer(30000, 1000) {
             override fun onTick(millisUntilFinished: Long) {}

             override fun onFinish() {
                 for (a in mClient) {
                     if (a.typeSerive) {
                         iSpam!!.spamMail(a.name)
                     }
                 }
                timer!!.start()
            }
        }
        timer!!.start()
    }
}